test_runner/fixtures/pageserver/utils.py

```diff
@@ -371,6 +373,32 @@ def list_prefix(
     return response
 
 
+def remote_storage_get_lastest_index_key(index_keys: List[str]) -> str:
+    """
+    Gets the latest index file key.
+
+    @param index_keys: A list of index keys of different generations.
+    """
+
+    def parse_gen(index_key: str) -> int:
+        parts = index_key.split("index_part.json-")
+        return int(parts[-1], base=16) if len(parts) == 2 else -1
+
+    return max(index_keys, key=parse_gen)
+
+
+def remote_storage_download_index_part(remote: S3Storage, index_key: str) -> IndexPartDump:
```

Comment:
It would be reasonable to put these util methods into the S3Storage type, they'll be more discoverable there.